### PR TITLE
perf db layout & backwards input size calculation correction

### PR DESCRIPTION
**Pull request by cderb**
workaround for exporting correct layout
input checking for import_configs to match tensor layouts 
bounding output size calculation to positive number 
re-organizing default values for convolutions

### CODE REVIEW

File: tuna/miopen/subcmd/export_db.py [hunk 1]
```diff
@@ -456,6 +457,13 @@ def export_pdb(dbt, args):
       ins_cfg_id = cfg_map[cfg_entry.id]
     else:
       cfg_dict = get_cfg_dict(cfg_entry, cfg_entry.input_t)
+
+      #override cfg layout with fdb key layout
+      if perf_db_entry.fdb_key:
```
Comment (alexandraBara) [inline tuna/miopen/subcmd/export_db.py:462]: are we assuming that we imported configs with the wrong layout and so we overwrite it with the fdb key layout?

**Review by alexandraBara — APPROVED** [review]

File: tuna/miopen/subcmd/export_db.py [hunk 2]
```diff
@@ -456,6 +457,13 @@ def export_pdb(dbt, args):
...
+        fds, vals, _, _ = parse_pdb_key(perf_db_entry.fdb_key)
+        key_layout = vals[fds.index('out_layout')]
+        cfg_dict['layout'] = key_layout
```
Comment (JehandadKhan) [inline tuna/miopen/subcmd/export_db.py]: This should be an error!

File: tuna/miopen/subcmd/import_configs.py [hunk 3]
```diff
@@ -154,6 +154,10 @@ def parse_line(args: argparse.Namespace, line: str, counts: dict,
     driver = DriverBatchNorm(line, args.command)
   else:
     driver = DriverConvolution(line, args.command)
+    if not driver.out_layout == driver.in_layout == driver.fil_layout:
+      raise ValueError(
+          f"Non-matching layouts {driver.out_layout}, {driver.in_layout}, {driver.fil_layout}"
+      )
```
Comment (JehandadKhan) [inline tuna/miopen/subcmd/import_configs.py]: This might not always be true, since the filter layout for NHWC convolution is CNHW ?

Comment (JehandadKhan) [reply tuna/miopen/subcmd/import_configs.py]: Lets confirm by running some driver commands

**Review by JehandadKhan — CHANGES_REQUESTED** [review]

File: tuna/miopen/subcmd/export_db.py (hunk 1 above)
Comment (cderb) [reply tuna/miopen/subcmd/export_db.py:462]: There was a fin issue which was causing layouts other than the one specified by the config to be tuned.
changing this statement to raise an error instead.

Comment (reidkwja) [reply tuna/miopen/subcmd/export_db.py:462]: [codebase design question] why do we need to override cfg layout with fdb keys?

**Review by alexandraBara — APPROVED** [review]

**Review by JehandadKhan — APPROVED** [review]

**Pull request merged**